src/pymor/parameters/functionals.py

```diff
@@ -107,17 +104,16 @@ def __rsub__(self, other):
     def __mul__(self, other):
         if not isinstance(other, (Number, ParameterFunctional)):
             return NotImplemented
-        if self.name != 'ProductParameterFunctional':
+        if self.name != 'ProductParameterFunctional' or (self.name == 'LincombParameterFunctional'
+                                                         and not isinstance(self, LincombParameterFunctional)):
```

Comment:
I think it should be `ProductParameterFunctional` instead of `LincombParameterFunctional` here, right?
```suggestion
        if self.name != 'ProductParameterFunctional' or not isinstance(self, ProductParameterFunctional):
```

Reply:
Thanks

Reply:
No Problem, @sdrave was a bit faster with the other suggestions....